source/Server.Extensibility.Authentication.OpenIDConnect/Web/UserAuthenticatedAction.cs

```diff
@@ -197,7 +197,7 @@ UserCreateResult GetOrCreateUser(UserResource userResource, string[] groups, Can
         {
             var identityToMatch = NewIdentity(userResource);
 
-            var user = userStore.GetByIdentity(identityToMatch);
+            var user = userStore.GetByIdentity(identityToMatch).SingleOrDefault();
```

Comment:
While syntactically correct, I wonder if we could throw a more helpful exception here? I think our future selves and customers will thank us.

```suggestion
            var matchingUsers = userStore.GetByIdentity(identityToMatch);
            if (matchingUsers.Count() > 1)
                throw new Exception("There are multiple users with this identity. OpenID Connect identity providers do not support users with duplicate email addresses. Please remove any duplicate users, or make the email addresses unique.");
            var user = matchingUsers.Single();
```